traffic_ops/testing/api/v4/origins_test.go

```diff
@@ -374,3 +473,411 @@ func DeleteTestOrigins(t *testing.T) {
+	originByDs, _, err := TOSession.GetOrigins(opts)
+	if err != nil {
+		t.Errorf("cannot get Origin by DeliveryService ID: %v - alerts: %+v", err, originByDs.Alerts)
+	}
+	if len(originByDs.Response) < 1 {
+		t.Fatalf("Expected atleast one Origin for GET Origin by Delivery Service, but found %d", len(originByDs.Response))
+	}
+
+	//Get Origins by Cachegroup
+	cachegroupID := *originByName.Response[0].CachegroupID
+	opts.QueryParameters.Del("deliveryservice")
+	opts.QueryParameters.Set("cachegroup", strconv.Itoa(cachegroupID))
+	originByCg, _, err := TOSession.GetOrigins(opts)
+	if err != nil {
+		t.Errorf("cannot get Origin by Cachegroup ID: %v - alerts: %+v", err, originByCg.Alerts)
+	}
+	if len(originByCg.Response) < 1 {
+		t.Fatalf("Expected atleast one Origin for GET Origin by Cachegroups, but found %d", len(originByCg.Response))
+	}
+
+	//Get Origins by Coordinate
+	CoordinateID := *originByName.Response[0].CoordinateID
+	opts.QueryParameters.Del("cachegroup")
+	opts.QueryParameters.Set("coordinate", strconv.Itoa(CoordinateID))
+	originByCoordinate, _, err := TOSession.GetOrigins(opts)
+	if err != nil {
+		t.Errorf("cannot get Origin by Coordinate ID: %v - alerts: %+v", err, originByCoordinate.Alerts)
+	}
+	if len(originByCoordinate.Response) < 1 {
+		t.Fatalf("Expected atleast one Origin for GET Origin by Coordinate, but found %d", len(originByCoordinate.Response))
+	}
+
+	//Get Origins by Profile
+	profileId := *originByName.Response[0].ProfileID
+	opts.QueryParameters.Del("coordinate")
+	opts.QueryParameters.Set("profileId", strconv.Itoa(profileId))
+	originByProfile, _, err := TOSession.GetOrigins(opts)
+	if err != nil {
+		t.Errorf("cannot get Origin by Profile ID: %v - alerts: %+v", err, originByProfile.Alerts)
+	}
+	if len(originByProfile.Response) < 1 {
+		t.Fatalf("Expected atleast one Origin for GET Origin by Profile, but found %d", len(originByProfile.Response))
+	}
+
+	//Get Origins by Primary
+	isPrimary := *originByName.Response[0].IsPrimary
+	opts.QueryParameters.Del("profileId")
+	opts.QueryParameters.Set("isPrimary", strconv.FormatBool(isPrimary))
+	originByPrimary, _, err := TOSession.GetOrigins(opts)
+	if err != nil {
+		t.Errorf("cannot get Origin by Primary ID: %v - alerts: %+v", err, originByPrimary.Alerts)
+	}
+	if len(originByPrimary.Response) < 1 {
+		t.Fatalf("Expected atleast one Origin for GET Origin by Primary, but found %d", len(originByPrimary.Response))
+	}
+
+	//Get Origins by Tenant
+	tenant := *originByName.Response[0].TenantID
+	opts.QueryParameters.Del("isPrimary")
+	opts.QueryParameters.Set("tenant", strconv.Itoa(tenant))
+	originByTenant, _, err := TOSession.GetOrigins(opts)
+	if err != nil {
+		t.Errorf("cannot get Origin by Tenant ID: %v - alerts: %+v", err, originByTenant.Alerts)
+	}
+	if len(originByTenant.Response) < 1 {
+		t.Fatalf("Expected atleast one Origin for GET Origin by Tenant, but found %d", len(originByTenant.Response))
+	}
+}
+
+func GetTestOriginsByInvalidParams(t *testing.T) {
```

Comment:
Do we need to test negative cases for `name`, `primary` and `dsid`?

Reply:
I don't have negative tests for this scenario. I will add.

Reply:
Added negative test for the scenarios.